### PR TITLE
prefer use of new-style NumPy random API in tests

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR follows up on #5357, mainly removing residual use of `np.random.RandomState` from test classes. This changes here are mostly trivial, but I have a couple specific points that may be worth considering:

1.) Regarding `segmentation.slice` and `segmentation.quickshift`. For those two, there are a lot of tests which check specific label values as below:

https://github.com/scikit-image/scikit-image/blob/e4cc46927944a75cec1beaf740904ce5b2143b52/skimage/segmentation/tests/test_slic.py#L28-L31

Is it really important to check that the specific values occur in this order? If not, we could replace with something like just verifying there is only a single, unique label within each region (The correct overall number of labels is already checked separately)

```Python
assert np.unique(seg[:10, :10]).size == 1
```
etc.

2.) There is one residual use of `RandomState` used when performing `slic` with a mask. 

https://github.com/scikit-image/scikit-image/blob/e4cc46927944a75cec1beaf740904ce5b2143b52/skimage/segmentation/slic_superpixels.py#L38

There are a couple of test results that seem to assume the existing initial seed points based on the current RandomState seed, so if we really want to remove that we have to update those test cases with new expected results. I just added a comment to that effect rather than switching it here. Similarly for a specific seed used to compare to precomputed results in one of the `unsupervised_wiener` test cases.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
This PR currently is only for modernizing the random call styles in the tests and does not make any API or behavior changes. I think we can address unifying `random_state` vs. `random_seed` etc. in a separate PR.

**Comment by mkcor** [comment]
> 1) Is it really important to check that the specific values occur in this order?

No, indeed, since the *order* of labels as 'integers' (which they are not) doesn't bear any specific meaning. By the way, I'm in favour of closing this issue: #5049

> If not, we could replace with something like just verifying there is only a single, unique label within each region.

Absolutely!

> (The correct overall number of labels is already checked separately)

All good then.

**Comment by mkcor** [comment]
@grlee77 the merge conflict here was expected from merging #5220.

**Comment by mkcor** [comment]
> 2) There are a couple of test results that seem to assume the existing initial seed points based on the current RandomState seed, so if we really want to remove that we have to update those test cases with new expected results. I just added a comment to that effect rather than switching it here.

Oh, iirc #5357 did that (updating expected results) for a few tests. Ok, let's leave these remaining pieces to a subsequent PR.

**Comment by grlee77** [comment]
> Oh, iirc #5357 did that (updating expected results) for a few tests. Ok, let's leave these remaining pieces to a subsequent PR.

Perhaps a bigger concern in the case of `slic` is whether downstream projects or users were using a specific seed when calling `slic`. If we change to the new generator within that function, then they would suddenly get a different initialization. I don't think the old interface is going away, so I wasn't sure it was worth updating in this case.

**Review by alexdesiqueira — APPROVED** [review]
LGTM; thanks @grlee77!

File: skimage/metrics/tests/test_structural_similarity.py [hunk 1]
```diff
@@ -56,18 +53,16 @@ def test_structural_similarity_image():
 @pytest.mark.parametrize('seed', [1, 2, 3, 5, 8, 13])
 @pytest.mark.parametrize('dtype', [np.float16, np.float32, np.float64])
 def test_structural_similarity_grad(seed, dtype):
-    N = 30
+    N = 60
```
Comment (mkcor) [inline skimage/metrics/tests/test_structural_similarity.py:56]: Why this change, just out of curiosity? :slightly_smiling_face:

**Review by mkcor — APPROVED** [review]
